torchgeo/datasets/urban3d.py

```diff
@@ -0,0 +1,281 @@
+    dataset is a dataset for semantic/instance segmentation of building footprints in
+    2D RGB imagery and 3D digital surface & terrain models
+
+    Dataset features:
+
+    * 278 images with 0.5 m per pixel resolution (2,048-2,048 px)
+    * 157,0000 building instances
+    * 6 bands (RGB, DTM, DSM, normalized DSM (nDSM))
+    * RGB imagery taken by Maxar WorldView 2-3 satellites
+
+    Dataset classes:
+
+    0. Background
+    1. Building
+
+    Dataset format:
+
+    * images are three-channel tiffs
+    * digital surface and terrain models are one-channel tiffs
+    * instance and binary masks are one-channel tiffs
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/AIPR.2017.8457973
+
+    .. note::
+
+       This dataset can be downloaded using the following bash script:
+
+       .. code-block:: bash
+
+          mkdir 01-Provisional_Train 02-Provisional_Test 03-Sequestered_Test
+          aws s3 sync s3://spacenet-dataset/Hosted-Datasets/Urban_3D_Challenge/01-Provisional_Train/ 01-Provisional_Train/
+          aws s3 sync s3://spacenet-dataset/Hosted-Datasets/Urban_3D_Challenge/02-Provisional_Test/ 02-Provisional_Test/
+          aws s3 sync s3://spacenet-dataset/Hosted-Datasets/Urban_3D_Challenge/03-Sequestered_Test/ 03-Sequestered_Test/
+
+    .. versionadded:: 0.3
+    """  # noqa: E501
+
+    directories = {
+        "train": "01-Provisional_Train",
+        "val": "02-Provisional_Test",
+        "test": "03-Sequestered_Test",
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+    ) -> None:
+        """Initialize a new Urban3DChallenge dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+        """
+        assert split in self.directories
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self._verify()
+        self.files = self._load_files()
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        rgb = self._load_image(files["rgb"])
+        dtm = self._load_image(files["dtm"])
+        dsm = self._load_image(files["dsm"])
+        ndsm = dsm - dtm
```

Comment:
Is this something we generate ourselves or something the dataset authors suggest? If it's just a linear combination of dtm and dsm maybe we should leave it up to a custom transform to create this.

Reply:
Something we generate ourselves but looking at the solutions to the dataset it's used as input instead of the dtm or dsm. I can take it out and make it a transform if that's what you suggest. No problem with that.